### PR TITLE
add 'remove' for List

### PR DESCRIPTION
**Pull request by chrisvfritz**
Added a `remove` function for List, similar to [Haskell's `delete`](http://hackage.haskell.org/package/base-4.8.0.0/docs/Data-List.html#v:delete). Of course, `delete` is already taken in JavaScript, so I called it `remove` instead.

### CODE REVIEW

File: src/List.ls [hunk 1]
```diff
@@ -15,6 +15,15 @@ filter = (f, xs) -->
...
 
+remove = (el, xs) -->
+  i = elem-index el, xs
+  results = xs.slice!
+  if i is void
+    results
+  else
+    results.splice i, 1
+    results
+
```
Comment (vendethiel) [inline src/List.ls]:
```
remove = (el, xs) -->
  i = elem-index el, xs
  results = xs.slice!
  if i?
    results.splice i, 1
  results
```

File: src/List.ls [hunk 2]
```diff
@@ -15,6 +15,13 @@ filter = (f, xs) -->
 reject = (f, xs) -->
   [x for x in xs when not f x]
 
+remove = (el, xs) -->
+  i = elem-index el, xs
+  results = xs.slice!
+  if i?
+    results.splice i, 1
+  results
```
Comment (anko) [inline src/List.ls]: Replacing 20-23 with—

```ls
xs.slice!
  ..splice i, 1 if i?
```
—would avoid repeating `results` 3 times.

Comment (vendethiel) [inline src/List.ls]:
```
[00:57] <vendethiel> posted a comment. could use cascade, but meh.
```

Comment (anko) [inline src/List.ls]: >but meh

Yup, it's a [bike shed](https://en.wikipedia.org/wiki/Bikeshedding). :) I'm happy either way.

Comment (chrisvfritz) [inline src/List.ls]: I'm happy to use the fancy cascade. :smiley: I generally avoid them in my own code, as they seem to confuse people coming from both imperative *and* functional backgrounds, but they are undeniably cool!

**Comment by chrisvfritz** [comment]
Anything else I should do to help get this merged in? Would love to DRY up some of my projects. :smiley:

**Comment by gkz** [comment]
thanks

**Pull request merged**